spec/lib/rbac/filterer_spec.rb

```diff
@@ -1147,20 +1147,53 @@ def combine_filtered_ids(user_filtered_ids, belongsto_filtered_ids, managed_filt
           :max_disk_usage_rate_average     => {},
           :min_net_usage_rate_average      => {},
           :max_net_usage_rate_average      => {},
-          :v_derived_storage_used          => {}
+          :v_derived_storage_used          => {},
+          :resource                        => {}
         }
       end
 
       # NOTE:  Think long and hard before you consider removing this test.
       # Many-a-hours wasted here determining this bug that resulted in
       # re-adding this test again.
-      it "should not raise an error when a polymorphic reflection is included" do
-        result = nil
+      #
+      # 2nd NOTE:  I did think (and wrote the above comment as well), however,
+      # now it seems we DO NOT SUPPORT polymorphic code, since we removed it
+      # here:
+      #
+      #   https://github.com/ManageIQ/manageiq/commit/8cc2277b
+      #
+      # That said, we should make sure this is erroring and no other callers
+      # are trying to do this (example:  MiqReport), so make sure this raises
+      # an error to inform the caller that fixing needs to happen.
+      #
+      it "raises an error when a polymorphic reflection is included and referenced" do
+        # NOTE: Fails if :references is passed with a value, or with no key,
+        # which it uses :include_for_find as the default.
         expect do
-          result = described_class.search :class            => "MetricRollup",
-                                          :include_for_find => @include
+          described_class.search :class            => "MetricRollup",
+                                 :include_for_find => @include,
+                                 :references       => @include
+        end.to raise_error(Rbac::PolymorphicError)
+
+        expect do
+          described_class.search :class            => "MetricRollup",
+                                 :include_for_find => @include
+        end.to raise_error(Rbac::PolymorphicError)
+      end
+
+      it "does not raise an error when a polymorphic reflection is only included" do
+        # NOTE:  if references is passed in, but is blank, then it is fine
+        expect do
+          described_class.search :class            => "MetricRollup",
+                                 :include_for_find => @include,
+                                 :references       => {}
+        end.not_to raise_error
+
+        expect do
+          described_class.search :class            => "MetricRollup",
+                                 :include_for_find => @include,
```

Comment:
This helps clarify it but is passing just the polymorphic in the references also supported?

Reply:
To avoid a 📖 explaining the Rails internals, the short answer is that `.references` does not work without `.includes`, per the docs:

https://api.rubyonrails.org/v5.1.7/classes/ActiveRecord/QueryMethods.html#method-i-references

So doing a test by itself would be pointless.

`ActiveRecord::Base.includes`, on the otherhand, under the hood will internally do a polymorphic check and decides whether or not to do one or multiple queries (the "one query" will do some joins as a result):

```ruby
MetricRollup.inclMetricRollup.includes(:time_profile => {})
# MetricRollup Load (2.2ms)  SELECT "metric_rollups".* FROM "metric_rollups"
# MetricRollup Inst Including Associations (0.1ms - 1rows)
# TimeProfile Load (0.3ms)  SELECT "time_profiles".* FROM "time_profiles" WHERE "time_profiles"."id" = 22000000000118
# TimeProfile Inst Including Associations (0.2ms - 1rows)
MetricRollup.includes(:time_profile => {}).references(:time_profiles)
# SQL (2.3ms)  SELECT "metric_rollups"."id" AS t0_r0,
#                     "metric_rollups"."timestamp" AS t0_r1,
#                     "metric_rollups"."capture_interval" AS t0_r2,
#                     "metric_rollups"."resource_type" AS t0_r3,
#                     ...
#                     "time_profiles"."id" AS t1_r0,
#                     "time_profiles"."description" AS t1_r1,
#                     "time_profiles"."profile_type" AS t1_r2,
#                     "time_profiles"."profile_key" AS t1_r3,
#                     LEFT OUTER JOIN "time_profiles" ON "time_profiles"."id" = "metric_rollups"."time_profile_id"
# MetricRollup Inst Including Associations (0.2ms - 1rows)
```

Guess this is a book...

* * *

But if you want a book, you can read the commit where I added this spec in the first place:  https://github.com/NickLaMuro/manageiq/commit/03284cf578110215524633633d7f652c2f2847c7

Reply:
lol. yup :book: says you need all `references() values in the `includes()`. ;)